message_ix_models/model/water/data/water_for_ppl.py

```diff
@@ -31,19 +35,26 @@ def missing_tech(x: pd.Series) -> pd.Series:
         "nuc_hc": 1 / 0.326,
         "nuc_lc": 1 / 0.326,
         "solar_th_ppl": 1 / 0.385,
+        "csp_sm1_res": 1 / 0.385,
+        "csp_sm3_res": 1 / 0.385,
     }
 
-    if pd.notna(x["technology"]) and x["technology"] in data_dic:
-        value = data_dic.get(x["technology"])
-        if x["value"] < 1:
-            value = max(x["value"], value)
-        # for backwards compatibility
-        if x["level"] == "cooling":
-            return pd.Series({"value": value, "level": "dummy_supply"})
-        else:
-            return pd.Series({"value": value, "level": x["level"]})
-    else:
-        return pd.Series({"value": x["value"], "level": x["level"]})
+    if pd.notna(x["technology"]):
+        # Find a matching key in `data_dic` using substring matching
+        matched_key = next((key for key in data_dic if key in x["technology"]), None)
+
+        if matched_key:
+            value = data_dic[matched_key]
+            if x["value"] < 1:
+                value = max(x["value"], value)
+            # for backwards compatibility
+            if x["level"] == "cooling":
+                return pd.Series({"value": value, "level": "dummy_supply"})
+            else:
+                return pd.Series({"value": value, "level": x["level"]})
```

Comment:
```suggestion
            return pd.Series({"value": value, "level": "dummy_supply"}) if x["level"] == "cooling" else pd.Series({"value": value, "level": x["level"]})
```
This avoids complexity and the need for a test case, but needs to get ruff applied to look good.

Reply:
Complexity as measured by code quality tools, yes. Writing is as I suggested enables the code to understand that all you're doing is returning a `pd.Series`, even though the contents differ. Writing it as you did means the tests (and potential readers) will have to check two arms of in if branch to see what you return.